src/main/java/org/springframework/integration/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -142,12 +142,21 @@ public void setOffsetManager(OffsetManager offsetManager) {
 		this.offsetManager = offsetManager;
 	}
 
-	public MessageListener getMessageListener() {
+	public Object getMessageListener() {
 		return messageListener;
 	}
 
-	public void setMessageListener(MessageListener messageListener) {
-		this.messageListener = messageListener;
+	public void setMessageListener(Object messageListener) {
+		if (autoCommitOffset) {
```

Comment:
Having this check in setter doesn't guarantee that `setAutoCommitOffset` will be invoked latter.
That's why I suggest to get rid of it at all.